### PR TITLE
1076_deprecate: deprecate questioned functions (and fix some bugs)

### LINKED ISSUE #1076: Decision on which "questioning" functions to deprecate?
### Feature Idea

See Stefan's comment in #936. We have lots of questioning badge functions that it might be nice to clean up for this release and decide to either keep or deprecate. Here's a proposal from my perspective and @bms63 & @thomas-neitmann feel free to make the final call:

- derive_var_ady - we should deprecate in favour of derive_vars_dy as we mention this simple example even in our new manifesto
- derive_var_astdy - as above
- derive_var_aendy - as above
- derive_var_trtedtm - personally i would leave it, as i think it's fine to have a more targeted function for something as common as this to make it easier for users to not have to figure out all the inputs to the more generic function like derive_vars_merged_dtm. So I would say not to deprecate. I see the difference with the above study day cases being that study day is so much simpler & i like that the filter_ex default gives you a starter for common code you could use to filter out invalid doses.
- derive_var_trtsdtm - as above
- derive_var_disposition_dt - i personally would deprecate in favour of derive_vars_merged_dt now, as we basically have 2 quite generic functions very similar but one is only for any DS date whereas the other can be used on any domain
- derive_var_lstalvdt - already deprecated in devel via https://github.com/pharmaverse/admiral/pull/959 in favour of derive_var_extreme_dt

Remember the impact to the ADaM vignettes and templates.

If a new contributor picks this up later once decision made, then reminder to use: https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1

### Relevant Input

NA

### Relevant Output

NA

### Reproducible Example/Pseudo Code

NA

### PR DESCRIPTION
**Pull request by bundfussr**
Closes #1076 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by bundfussr** [comment]
@pharmaverse/admiral , has anyone time to review this pull request?

**Comment by bundfussr** [comment]
@bms63 , thanks, I have fixed the merge conflicts.

File: R/derive_var_ady.R [hunk 1]
```diff
@@ -38,22 +40,15 @@
 #'
 #' @export
 #'
-#' @examples
```
Comment (bms63) [inline R/derive_var_ady.R:41]: Should we put this in the deprecation strategy to remove examples from deprecated functions?

**Comment by millerg23** [comment]
@bundfussr I get same ERRORs in "Man Pages" and "Style" in my pull request also! https://github.com/pharmaverse/admiral/pull/1111

File: R/derive_var_ady.R (hunk 1 above)
Comment (bundfussr) [reply R/derive_var_ady.R:41]: I put it there.

File: tests/testthat/test-deprecation.R [hunk 2]
```diff
@@ -571,3 +537,122 @@ test_that("a warning is issued when using `derive_var_worst_flag()` with `filter
...
+  )
+
+  expect_warning(
+    derive_var_ady(input),
+    "deprecated",
+    fixed = TRUE
+  )
+})
+
+test_that("a warning is issued when using `derive_var_enady()`", {
```
Comment (bms63) [inline tests/testthat/test-deprecation.R:556]: function is misspelled

Comment (bms63) [reply tests/testthat/test-deprecation.R:556]: derive_var_aendy

File: vignettes/admiral.Rmd [hunk 3]
```diff
@@ -42,8 +42,32 @@ ex <- admiral_ex
...
 adsl <- dm %>%
   mutate(TRT01P = ARM, TRT01A = ACTARM) %>%
-  derive_var_trtsdtm(dataset_ex = admiral_ex) %>%
-  derive_var_trtedtm(dataset_ex = admiral_ex) %>%
+  derive_vars_merged_dtm(
+    dataset_add = ex,
+    filter_add = (EXDOSE > 0 |
+                    (EXDOSE == 0 &
+                       str_detect(EXTRT, "PLACEBO"))) &
+      nchar(EXSTDTC) >= 10,
```
Comment (bms63) [inline vignettes/admiral.Rmd:50]: Do we need this filter for these beginner vignettes?  Just a lot to digest on the intro level.  

If so, maybe we should add in a few Notes about what is happening here??

Comment (bms63) [reply vignettes/admiral.Rmd:50]: We do talk about the filter parameter later in the Derive Disposition variables  section

**Review by bms63 — APPROVED** [review]
Hey Stefan.  Just concern around this filtering statement.

File: R/derive_var_ady.R (hunk 1 above)
Comment (bms63) [reply R/derive_var_ady.R:41]: Oh lol I was looking at deprecation strategy for 0.6.3 website and not the updated 0.7.0 website.  Thanks!

File: vignettes/admiral.Rmd (hunk 3 above)
Comment (bundfussr) [reply vignettes/admiral.Rmd]: I have simplified the condition.

File: vignettes/adsl.Rmd [hunk 4]
```diff
@@ -98,18 +98,42 @@ adsl <- dm %>%
...
 ```{r eval=TRUE}
 adsl <- adsl %>%
-  derive_var_trtsdtm(dataset_ex = ex) %>%
-  derive_var_trtedtm(dataset_ex = ex)
+  derive_vars_merged_dtm(
+    dataset_add = ex,
+    filter_add = (EXDOSE > 0 |
+                    (EXDOSE == 0 &
+                       str_detect(EXTRT, "PLACEBO"))) & nchar(EXSTDTC) >= 10,
+    new_vars_prefix = "TRTS",
```
Comment (bms63) [inline vignettes/adsl.Rmd:113]: Do we want to simplify this one one as well?

**Review by bms63 — APPROVED** [review]

File: vignettes/adsl.Rmd (hunk 4 above)
Comment (bms63) [reply vignettes/adsl.Rmd:113]: Love it!

**Review by bms63 — APPROVED** [review]
I'm happy with this.  Do we want another reviewer or we can just merge in?

**Comment by bundfussr** [comment]
From my side you can merge it.

**Comment by bms63** [comment]
Oh I just noticed the 1056 commit messages are using incorrect issue link.  Just FYI - it should be #1056 
image

**Pull request merged**